src/backend/jitsi_magnify/core/api.py

```diff
@@ -32,42 +37,73 @@ def exception_handler(exc, context):
+        "user_id": user.id,
+        "exp": expiration,
+        "iat": datetime.datetime.utcnow(),
+        "moderator": user.is_staff,
+        "aud": "jitsi",
+        "iss": settings.JWT_CONFIGURATION.get("jitsi_app_id"),
+        "sub": "meet.jitsi",
+        "room": room,
+    }
+
+    jitsi_user = {
+        "avatar": settings.JWT_CONFIGURATION.get("guest_avatar"),
+        "name": user.username,
+        "email": user.email,
+    }
 
-        # Add custom claims
-        jitsiUser = {'avatar': settings.JWT_CONFIGURATION.get("guest_avatar"),
-                'name': user.username,
-                'email': user.email}
+    token_payload["context"] = {"user": jitsi_user}
 
-        token['context'] = {'user': jitsiUser}
+    return token_payload
 
-        token['moderator'] = user.is_staff
-        token['aud'] = "jitsi"
-        token['iss'] = settings.JWT_CONFIGURATION.get("jitsi_app_id")
-        token['sub'] = "meet.jitsi"
-        token['room'] = room
 
-        return token
+def generate_token(user, room, expiration):
+    """Generate the access token that will give access to the room"""
+    token_payload = create_payload(user, room, expiration=expiration)
+    token = jwt.encode(
+        token_payload,
+        settings.JWT_CONFIGURATION.get("jitsi_secret_key"),
+        algorithm="HS256",
+    )
 
-class CustomTokenView(TokenObtainPairView):
-    """View for token creation"""
-    serializer_class = CustomTokenSerializer
+    return token
 
-def obtainTokenView(request, room):
+
+@api_view(("GET",))
+def obtain_token_view(request, room):
     """View for token fetching based on a guest user and a specific room
     This redirects to jitsi with the token"""
     user, created = User.objects.get_or_create(username="guest")
-    
     if created:
         user.set_password(settings.JWT_CONFIGURATION.get("guest_default_password"))
         user.save()
 
-    token = CustomTokenSerializer.get_token(user=user, room=room)
-
-    return HttpResponseRedirect(settings.JWT_CONFIGURATION.get("jitsi_url") + "/" + room + "?jwt=" + str(token))
+    access_token_expiration = datetime.datetime.utcnow() + datetime.timedelta(minutes=5)
+    refresh_token_expiration = datetime.datetime.utcnow() + datetime.timedelta(days=7)
+    access_token = generate_token(user, room, access_token_expiration)
+    refresh_token = generate_token(user, room, refresh_token_expiration)
+
+    base_url = settings.JWT_CONFIGURATION.get("jitsi_domain_name")
+    url_params = {"jwt": access_token}
+    encoded_params = urlencode(url_params)
+    url = f"https://{base_url}/{room}?{encoded_params}"
+
+    # As we allow users to pass information to redirect url (with the parameter room),
+    # we check that this url is safe before redirecting
+    if not is_safe_url(url, allowed_hosts=base_url, require_https=True):
+        return Response(f"Redirection to url {url} is not allowed", status=404)
+
+    response = HttpResponseRedirect(url)
+    response.set_cookie(
```

Comment:
not sure how this cookie is used?
I found another dependency to simple jwt here https://github.com/openfun/jitsi-magnify/blob/main/src/backend/jitsi_magnify/settings.py#L183 :wink:

Reply:
It is not used yet, but tokens can usually be refreshed with a route such as /api/token/refresh, with the refresh token that is kept in local storage (here in cookies). I am still documenting myself on this but it should be implemented in the future

Reply:
Well... actually we don't need that because we will generate a fresh JWT token for you anytime you call this view... refreshing the token is required when it is used as the main authentication. Right?
If you agree, then we can remove this?

Reply:
It should not be required here, yes, so I will remove everything related to the refresh token. We'll have to test once jitsi-magnify is deployed that there are no problems once access_token is expired (such as accessing other participants' videos, muting, etc...), however we think that as soon as it connects to the web socket for jitsi, it does not need the token anymore